File: offer/63/jphoebe.java
import java.util.ArrayList;
import java.util.List;
public class Solution {

    List<Integer> list = new ArrayList<>();

    public void Insert(Integer num) {

        if (list.isEmpty()){
            list.add(num);
        }else{
            if (list.get(0) >= num){
                list.add(0, num);
            }else if (list.get(list.size()-1)<=num){
                list.add(num);
            }else{
                for (int i = 0; i < list.size()-1; i++) {
                    Integer current = list.get(i);
                    Integer next = list.get(i+1);
                    if (current <= num && num<=next){
                        list.add(i+1, num);
                        return ;
                    }
                }
            }
        }
    }

    public Double GetMedian() {
        if (this.list.size()<=0){
            return null;
        }
        int midd = list.size() / 2;
        if (list.size() % 2 == 0) {
            double result =  (list.get(midd) + list.get(midd-1)) / 2.0;
            return result;
        } else {
            return Double.valueOf(list.get(midd));
        }
    }


}

File: offer/64/JPhoebe.py
import java.util.ArrayList;
public class Solution {
    public ArrayList<Integer> maxInWindows(int [] num, int size)
    {
        if (num == null || num.length == 0 || size == 0 || num.length < size) {
            return new ArrayList<>();
        }
        ArrayList<Integer> result = new ArrayList<>();
        Integer windowMaxValue = null;
        for (int i = size-1; i < num.length; i++) {
            if (windowMaxValue == null) {
                windowMaxValue = num[i];
            }
            for (int j = 1; j < size; j++) {
                if (num[i-j] > windowMaxValue) {
                    windowMaxValue = num[i-j];
                }
            }
            result.add(windowMaxValue);
            windowMaxValue = null;
        }
        return result;
    }
}
